### DISCUSSION TITLE
Is there a way to send data/events to JavaScript from Julia?

### DISCUSSION THREAD

**Discussion by Nufflee** [fonsp/Pluto.jl#1110] [Questions about using Pluto] [answered]
I know it's possible to send data from JavaScript to Julia using `@bind` and input events but could I do it the other way around? My goal is to write a little Julia library for drawing on an HTML5 Canvas but I need to be able to send events to JavaScript in order to do that.

**Comment by fonsp** [comment, marked as answer]
This is not the other way around, it is exactly what `@bind` is for! https://github.com/fonsp/Pluto.jl/issues/598

**Reply by Nufflee** [reply]
I may be missing something or not describing my problem well enough but I already saw that issue and what's being shown there is the opposite of what I need. They're binding a Julia array to a JS context so that data drawn on the canvas by the user can be transferred into the Julia array. What I need is transferring a drawing/data _from_ a Julia array _to_ the JS context and drawing it on the canvas, no user input involved.

It does seem like `PlutoRunner.publish_to_js` may be what I'm looking for but I haven't got a chance to try it out yet.

**Reply by fonsp** [reply]
I see! in that case PlutoRunner.publish_to_js or just JSON.jl is what you want. Take a look at https://github.com/fonsp/Pluto.jl/pull/1124 and the Javascript sample notebook inside Pluto
